packages/mui-base/src/Unstable_TrapFocus/Unstable_TrapFocus.js

```diff
@@ -183,14 +184,14 @@ function Unstable_TrapFocus(props) {
         // Once IE11 support is dropped the focus() call can be unconditional.
         if (nodeToRestore.current && nodeToRestore.current.focus) {
           ignoreNextEnforceFocus.current = true;
-          nodeToRestore.current.focus();
+          nodeToRestore.current.focus({ preventScroll: preventScrollOnFocusRestoration });
         }
 
         nodeToRestore.current = null;
       }
     };
-    // Missing `disableRestoreFocus` which is fine.
-    // We don't support changing that prop on an open TrapFocus
+    // Missing `disableRestoreFocus` / `preventScrollOnFocusRestoration` which is fine.
```

Comment:
Why not allow changing it during the lifetime of TrapFocus?

Reply:
If I add it to the dependencies of the `useEffect`, the cleanup function will run when the user change  `preventScrollOnFocusRestoration` value, which is not what we want right ?

Reply:
We could store the prop value in a ref. It won't have to be added to the dependency array this way.

Reply:
The same thing could be done for `disableRestoreFocus` or is there another reason we can't support changing `disableRestoreFocus` ?

Reply:
I'm not aware of any TBH. I see it was introduced by @eps1lon in #22155 so perhaps there is a need for it.